docs/unread_messages.md

```diff
@@ -0,0 +1,50 @@
+# Unread message synchronization
+
+In general displaying unread counts for all streams and topics may require
+downloading an unbounded number of messages. Consider a user who has a muted
+stream or topic and has not read the backlog in a month; to have an accurate
+unread count we would need to load all messages this user has received in the
+past month. This is inefficient for web clients and even more for mobile
+devices.
+
+We work around this by including a list of unread message ids in the initial
+state grouped by relevant conversation keys. This data is included in the
+`unread_msgs` key if both `update_message_flags` and `message` are required
+in the register call.
+
+```
+{
+    "huddles": [
+        {
+            "user_ids_string": "3,4,6",
+            "unread_message_ids": [
+                34
+            ]
+        }
+    ],
+    "streams": [
+        {
+            "stream_id": 1,
+            "topic": "test",
+            "unread_message_ids": [
+                33
+            ]
+        }
+    ],
+    "pms": [
+        {
+            "sender_id": 3,
+            "unread_message_ids": [
+                31,
+                32
+            ]
```

Comment:
I think it might be simpler for how I expect clients to use these data if we just had this have a `user_ids_string` as well, even though it's a 1:1 PM?  And possibly combined this dict with "huddles"?  

My reasoning is that probably clients will be rendering a view that just shows all threads with unread messages, regardless of type, sorted by the most recent message ID (or something).  

I guess a related question is whether the top-level streams/pms/huddles split is necessary or we should just have one list, and you determine the type by checking for the `stream_id` key.